$NetBSD$

On NetBSD, don't use -ldl (why?).

For uuid, insist on library, not just header (why?).

Omit ldconfig (from OpenBSD).

--- src/CMakeLists.txt.orig	2017-02-20 22:06:57.000000000 +0000
+++ src/CMakeLists.txt	2017-02-25 12:18:04.360077118 +0000
@@ -95,6 +95,8 @@
 if (UNIX)
 	if (APPLE)
 		set (MOSQ_LIBS ${MOSQ_LIBS} dl m)
+	elseif (${CMAKE_SYSTEM_NAME} MATCHES "NetBSD")
+		set (MOSQ_LIBS ${MOSQ_LIBS} m)
 	else (APPLE)
 		set (MOSQ_LIBS ${MOSQ_LIBS} dl m)
 		find_library(LIBRT rt)
@@ -115,10 +117,11 @@
 # Simple detect libuuid
 if(NOT APPLE)
 	FIND_PATH(UUID_HEADER uuid/uuid.h)
-	if (UUID_HEADER)
+	FIND_LIBRARY(UUID_LIBRARY NAMES uuid)
+	IF (UUID_HEADER AND UUID_LIBRARY)
 		add_definitions(-DWITH_UUID)
-		set (MOSQ_LIBS ${MOSQ_LIBS} uuid)
-	endif (UUID_HEADER)
+		set (MOSQ_LIBS ${MOSQ_LIBS} ${UUID_LIBRARY})
+	endif ()
 endif(NOT APPLE)
 
 target_link_libraries(mosquitto ${MOSQ_LIBS})
